File: ios/NorthAx/Models/AthleteThresholds.swift
import Foundation

enum PaceUnit: String, Codable { case km, mile }
enum PoolUnit: String, Codable { case pool25m, pool50m, openWater }

/// Athlete physiological thresholds used at render time to compute training
/// zones (the GRAPH agent owns ZoneMath; this model just carries the data).
struct AthleteThresholds: Codable, Equatable {
    var ftpWatts: Int?
    var thresholdHr: Int?
    var maxHr: Int?
    var runThresholdPaceSecPerKm: Int?
    var paceUnit: PaceUnit
    var swimThresholdPaceSecPer100m: Int?
    var poolUnit: PoolUnit

    init(
        ftpWatts: Int? = nil,
        thresholdHr: Int? = nil,
        maxHr: Int? = nil,
        runThresholdPaceSecPerKm: Int? = nil,
        paceUnit: PaceUnit = .km,
        swimThresholdPaceSecPer100m: Int? = nil,
        poolUnit: PoolUnit = .pool25m
    ) {
        self.ftpWatts = ftpWatts
        self.thresholdHr = thresholdHr
        self.maxHr = maxHr
        self.runThresholdPaceSecPerKm = runThresholdPaceSecPerKm
        self.paceUnit = paceUnit
        self.swimThresholdPaceSecPer100m = swimThresholdPaceSecPer100m
        self.poolUnit = poolUnit
    }
}

File: android/app/src/main/java/app/northax/domain/model/AthleteThresholds.kt
package app.northax.domain.model

enum class PaceUnit(val raw: String) {
    Km("km"), Mile("mile");

    companion object {
        fun fromRaw(raw: String): PaceUnit? = entries.firstOrNull { it.raw == raw }
    }
}

enum class PoolUnit(val raw: String) {
    Pool25m("pool25m"), Pool50m("pool50m"), OpenWater("openWater");

    companion object {
        fun fromRaw(raw: String): PoolUnit? = entries.firstOrNull { it.raw == raw }
    }
}

/**
 * Athlete physiological thresholds used at render time to compute training
 * zones (ZoneMath consumes these; this model just carries the data).
 */
data class AthleteThresholds(
    val ftpWatts: Int? = null,
    val thresholdHr: Int? = null,
    val maxHr: Int? = null,
    val runThresholdPaceSecPerKm: Int? = null,
    val paceUnit: PaceUnit = PaceUnit.Km,
    val swimThresholdPaceSecPer100m: Int? = null,
    val poolUnit: PoolUnit = PoolUnit.Pool25m,
)
